drivers/sensor/bosch/bmp388/bmp388_spi.c

```diff
@@ -22,43 +22,31 @@ static int bmp388_bus_check_spi(const union bmp388_bus *bus)
 	return spi_is_ready_dt(&bus->spi) ? 0 : -ENODEV;
 }
 
-static int bmp388_reg_read_spi(const union bmp388_bus *bus,
-			       uint8_t start, uint8_t *buf, int size)
+static int bmp388_reg_read_spi(const union bmp388_bus *bus, uint8_t regaddr, uint8_t *buf, int size)
 {
-	uint8_t addr;
-	const struct spi_buf tx_buf = {
-		.buf = &addr,
-		.len = 1
+	int ret;
+	uint8_t buffer[size + 2];
```

Comment:
Hmmm - May we validate this array sizing against a reasonable limit? Or statically size this buffer assuming the worst case we'd ever read?

Trying to stay away from unconditionally adjusting sizing the buffer irrespective of the value `size` may have.

Reply:
Thanks, that is a very strong argument - I will add a validation, and do the rebase again - for the i2c.dtsi

Reply:
@MaureenHelm i2c.dtsi. I rebased twice because there was a commit 9d0486e3ee86 just a minute ago and the sequence numbering is broken gain.